### ISSUE TITLE
Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. 

### ISSUE THREAD

**Issue by Lasith650** [wso2/product-sp#796]
**Description**
```
[2018-09-10_16-56-01_705] ERROR {org.wso2.siddhi.core.stream.input.source.Source} - Error on 'CSVDefaultMapping'. Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. Error while connecting at Source 'file' at 'UsageStream2'. Will retry in '5 sec'. (Encoded)
org.wso2.siddhi.core.exception.ConnectionUnavailableException: Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. 
	at org.wso2.extension.siddhi.io.file.FileSource.deployServers(FileSource.java:544)
	at org.wso2.extension.siddhi.io.file.FileSource.connect(FileSource.java:370)
	at org.wso2.siddhi.core.stream.input.source.Source.connectWithRetry(Source.java:130)
	at org.wso2.siddhi.core.SiddhiAppRuntime.startSources(SiddhiAppRuntime.java:424)
	at org.wso2.siddhi.core.SiddhiAppRuntime.start(SiddhiAppRuntime.java:367)
	at org.wso2.carbon.siddhi.editor.core.internal.DebugRuntime.start(DebugRuntime.java:68)
	at org.wso2.carbon.siddhi.editor.core.internal.DebugProcessorService.start(DebugProcessorService.java:37)
	at org.wso2.carbon.siddhi.editor.core.internal.EditorMicroservice.start(EditorMicroservice.java:528)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.wso2.msf4j.internal.router.HttpMethodInfo.invokeResource(HttpMethodInfo.java:187)
	at org.wso2.msf4j.internal.router.HttpMethodInfo.invoke(HttpMethodInfo.java:143)
	at org.wso2.msf4j.internal.MSF4JHttpConnectorListener.dispatchMethod(MSF4JHttpConnectorListener.java:218)
	at org.wso2.msf4j.internal.MSF4JHttpConnectorListener.lambda$onMessage$57(MSF4JHttpConnectorListener.java:129)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.wso2.transport.remotefilesystem.exception.RemoteFileSystemConnectorException: Failed to initialize File server connector for Service: CSVDefaultMapping
	at org.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.(RemoteFileSystemServerConnectorImpl.java:53)
	at org.wso2.transport.remotefilesystem.impl.RemoteFileSystemConnectorFactoryImpl.createServerConnector(RemoteFileSystemConnectorFactoryImpl.java:40)
	at org.wso2.extension.siddhi.io.file.FileSource.deployServers(FileSource.java:538)
	... 18 more
Caused by: org.wso2.transport.remotefilesystem.exception.RemoteFileSystemConnectorException: [CSVDefaultMapping] File system server connector is used to listen to a folder. But the given path does not refer to a folder.
	at org.wso2.transport.remotefilesystem.server.RemoteFileSystemConsumer.(RemoteFileSystemConsumer.java:105)
	at org.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.(RemoteFileSystemServerConnectorImpl.java:51)
	... 20 more

[2018-09-10_16-56-06_707] ERROR {org.wso2.siddhi.core.stream.input.source.Source} - Error on 'CSVDefaultMapping'. Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. Error while connecting at Source 'file' at 'UsageStream2'. Will retry in '10 sec'. (Encoded)
org.wso2.siddhi.core.exception.ConnectionUnavailableException: Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. 
	at org.wso2.extension.siddhi.io.file.FileSource.deployServers(FileSource.java:544)
	at org.wso2.extension.siddhi.io.file.FileSource.connect(FileSource.java:370)
	at org.wso2.siddhi.core.stream.input.source.Source.connectWithRetry(Source.java:130)
	at org.wso2.siddhi.core.stream.input.source.Source$1.run(Source.java:142)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.wso2.transport.remotefilesystem.exception.RemoteFileSystemConnectorException: Failed to initialize File server connector for Service: CSVDefaultMapping
	at org.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.(RemoteFileSystemServerConnectorImpl.java:53)
	at org.wso2.transport.remotefilesystem.impl.RemoteFileSystemConnectorFactoryImpl.createServerConnector(RemoteFileSystemConnectorFactoryImpl.java:40)
	at org.wso2.extension.siddhi.io.file.FileSource.deployServers(FileSource.java:538)
	... 10 more
Caused by: org.wso2.transport.remotefilesystem.exception.RemoteFileSystemConnectorException: [CSVDefaultMapping] File system server connector is used to listen to a folder. But the given path does not refer to a folder.
	at org.wso2.transport.remotefilesystem.server.RemoteFileSystemConsumer.(RemoteFileSystemConsumer.java:105)
	at org.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.(RemoteFileSystemServerConnectorImpl.java:51)
	... 12 more

[2018-09-10_16-56-16_709] ERROR {org.wso2.siddhi.core.stream.input.source.Source} - Error on 'CSVDefaultMapping'. Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. Error while connecting at Source 'file' at 'UsageStream2'. Will retry in '15 sec'. (Encoded)
org.wso2.siddhi.core.exception.ConnectionUnavailableException: Failed to connect to the remote file system server through the siddhi app 'CSVDefaultMapping'. 
	at org.wso2.extension.siddhi.io.file.FileSource.deployServers(FileSource.java:544)
	at org.wso2.extension.siddhi.io.file.FileSource.connect(FileSource.java:370)
	at org.wso2.siddhi.core.stream.input.source.Source.connectWithRetry(Source.java:130)
	at org.wso2.siddhi.core.stream.input.source.Source$1.run(Source.java:142)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.wso2.transport.remotefilesystem.exception.RemoteFileSystemConnectorException: Failed to initialize File server connector for Service: CSVDefaultMapping
	at org.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.(RemoteFileSystemServerConnectorImpl.java:53)
	at org.wso2.transport.remotefilesystem.impl.RemoteFileSystemConnectorFactoryImpl.createServerConnector(RemoteFileSystemConnectorFactoryImpl.java:40)
	at org.wso2.extension.siddhi.io.file.FileSource.deployServers(FileSource.java:538)
	... 10 more
Caused by: org.wso2.transport.remotefilesystem.exception.RemoteFileSystemConnectorException: [CSVDefaultMapping] File system server connector is used to listen to a folder. But the given path does not refer to a folder.
	at org.wso2.transport.remotefilesystem.server.RemoteFileSystemConsumer.(RemoteFileSystemConsumer.java:105)
	at org.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.(RemoteFileSystemServerConnectorImpl.java:51)
	... 12 more
```

**Affected Product Version:**
wso2sp-4.3.0-rc1

  
**Steps to reproduce:**
In the prerequisites section of the sample CSVDefaultMapping it mentions to :  

 "Edit the uri '{WSO2SPHome}/samples/artifacts/CSVMappingWithFile/new/example.csv' by replacing {WSO2SPHome} with the absolute path of your WSO2SP home directory. You can also change the path after {WSO2SPHome} (/samples/artifacts/CSVMappingWithFile/new/) if you want to publish your event file to a different location."

This gives an error. If the uri is given as   {WSO2SPHome}/samples/artifacts/CSVMappingWithFile/new,
the siddhi application starts successfully while displaying that the deployment was successful.

**Comment by minudika** [comment]
This is due to some invalid instructions given in the sample. It has been fixed and will be available in the 4.3.0 RC2.

**Comment by minudika** [comment]
@Niveathika92  yes this is available in SP 4.3.0 RC2 and later releases
